### ISSUE TITLE
nsqd: documentation for dynamic configuration

### ISSUE THREAD

**Issue by harlow** [nsqio/nsq#708]
A PR was merged to expose a new `/config` endpoint in https://github.com/nsqio/nsq/pull/601.
https://github.com/nsqio/nsq/blob/master/nsqd/http.go#L67

I was wondering if there is any documentation on how to leverage the endpoint in a distributed system?

**Comment by harlow** [comment]
cc @paddyforan mind sharing how you are using the endpoint? 

I'm using Fleet to schedule units on CoreOS cluster. With this setup would I need to run a "sidekick service" alongside `nsqlookupd` that sends PUT requests to all of the `nsqd` services across the cluster each time a new `nsqlookupd` is brought up?

**Comment by mreiferson** [comment]
@harlow I'm not sure of the specifics, but the idea was that this HTTP API allowed for "shim" services to be built that dealt with the specifics of interacting with your dynamic service discovery system and could make state changes to a local `nsqd`, so I think the answer to your question is "yes".

If you've started to work through this and are interested in contributing some formal docs, that would be awesome!

You can update documentation at https://github.com/nsq/nsqio.github.io

**Comment by harlow** [comment]
@mreiferson thanks for reply. I've got NSQ running in our cluster w/ a sidecar discovery service for `nsqlookupd`

The process:
1. launch nsqlookupd w/ the discovery sidecar
2. nsqlookupd sidecar puts the IP address of the current server into etcd
3. when nsqd launches they register themselves with each of the nsqlookupd IPs from etcd

This works well but suffers from the issue that if an `nsqlookupd` unit were moved to another box the `nsqd` instances would have an invalid IP and would need to be restarted.

The `/config` endpoint would help to solve this issue; when a `nsqlookupd` is moved between boxes we'd need to send updated IP's to the `config` endpoint.

Would be really nice to add DNS discovery to the `nsqd` services; This way we could just add one address and they could periodically check DNS and just update their IP's with A-Record info.

![screen shot 2016-01-20 at 10 51 26 am](https://cloud.githubusercontent.com/assets/739782/12459675/d252922e-bf64-11e5-9043-e37bbeaf6836.png)
